content/post/walkthrough-exploiting-bugs-to-gain-code-execution.md

```diff
@@ -0,0 +1,338 @@
++++
+author = "cbrown"
+date = "2015-09-18T19:55:54-07:00"
+description = ""
+tags = ["exploit", "rop", "aslr", "shell"]
+title = "Walkthrough: Exploiting bugs to gain code execution on a modern operating system"
+
++++
+
+Whenever I discover a bug or typo in code I've written, I like to ask myself what the worst case effects of that typo might be. The most interesting question of course, is "If someone else found this before I did, could they have used it to gain arbitrary code execution?"
```

Comment:
I believe `worst case` should be hyphenated. I'm having trouble finding something definitive, but I _think_ it's correct.